File: C/Complex.c
#include <stdio.h>

struct complex
{
    float real;
    float imaginary;
};

typedef struct complex COMPLEX;

COMPLEX sum(COMPLEX c1, COMPLEX c2);
void display(COMPLEX c);

int main()
{
    COMPLEX c1, c2, c3;
    printf("Enter a complex number (r, i): ");
    scanf("%f, %f", &c1.real, &c1.imaginary);
    printf("Enter another complex number (r, i): ");
    scanf("%f, %f", &c2.real, &c2.imaginary);
    c3 = sum(c1, c2);
    printf("c1: ");
    display(c1);
    printf("c2: ");
    display(c2);
    printf("c3 (sum): ");
    display(c3);
    return 0;
}

COMPLEX sum(COMPLEX c1, COMPLEX c2)
{
    COMPLEX c3;
    c3.real = c1.real + c2.real;
    c3.imaginary = c1.imaginary + c2.imaginary;
    return c3;
}

void display(COMPLEX c)
{
    printf("real: %1.1f, imaginary: %1.1f\n", c.real, c.imaginary);
}

File: Java/nopackage/Complex.java
import java.util.Scanner;

class Complex {
    private float real;
    private float imaginary;

    public void getData() {
        Scanner scan = new Scanner(System.in);
        System.out.print("real: ");
        real = scan.nextFloat();
        System.out.print("imaginary: ");
        imaginary = scan.nextFloat();
    }

    public Complex sum(Complex c) {
        Complex t = new Complex();
        t.real = real + c.real;
        t.imaginary = imaginary + c.imaginary;
        return t;
    }

    public void putData() {
        System.out.println("real: " + real + ", imaginary: " + imaginary);
    }

    public static void main(String[] args) {
        Complex c1 = new Complex(); //Dynamic creation of objects
        Complex c2 = new Complex();
        Complex c3 = new Complex();
        System.out.println("Enter a complex number:");
        c1.getData();
        System.out.println("Enter another complex number:");
        c2. getData();
        c3 = c1.sum(c2);
        System.out.print("c1: ");
        c1.putData();
        System.out.print("c2: ");
        c2.putData();
        System.out.print("c3 (sum): ");
        c3.putData();
    }
}